app/code/Magento/QuoteGraphQl/Model/Cart/AddToCartHandlerResolver.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteGraphQl\Model\Cart;
+
+/**
+ * @inheritdoc
+ */
+class AddToCartHandlerResolver
```

Comment:
It should be an interface with `product type` and `cart item data` (under the hood will list of handlers)